packages/std/src/testing/mock.rs

```diff
@@ -126,32 +127,30 @@ impl Api for MockApi {
     }
 
     fn addr_canonicalize(&self, input: &str) -> StdResult<CanonicalAddr> {
-        match decode(input) {
-            Ok((prefix, _, _)) if prefix != self.bech32_prefix => {
-                Err(StdError::generic_err("Wrong bech32 prefix"))
-            }
-            Ok((_, _, Variant::Bech32m)) => Err(StdError::generic_err("Wrong bech32 variant")),
-            Err(_) => Err(StdError::generic_err("Error decoding bech32")),
-            Ok((_, decoded, Variant::Bech32)) => match Vec::<u8>::from_base32(&decoded) {
-                Ok(bytes) => {
-                    validate_length(&bytes)?;
-                    Ok(bytes.into())
-                }
-                Err(_) => Err(StdError::generic_err("Invalid bech32 data")),
-            },
+        let hrp_str = CheckedHrpstring::new::<Bech32>(input)
+            .map_err(|_| StdError::generic_err("Error decoding bech32"))?;
+
+        if !hrp_str
+            .hrp()
+            .as_bytes()
+            .eq_ignore_ascii_case(self.bech32_prefix.as_bytes())
```

Comment:
It was actually fixing failing test cases, apparently `decode` normalized them into *some* casing.  
We have these test cases and they started failing:

https://github.com/CosmWasm/cosmwasm/blob/092d222169cf1aae4ef4b14ba1fc018d2d50a734/packages/std/src/testing/mock.rs#L1172-L1193

Since we can't use `decode` anymore (because it internally tries to use either, `Bech32` and `Bech32m`, without a marker which format was used. Was one of the breaking changes), so we had to drop down to the low level API.

Reply:
Ah okay, perfect. Then we keep the same behaviour as before. Thanks for clarifying.